src/applications/travel-pay/containers/TravelPayStatusApp.jsx

```diff
@@ -48,9 +49,18 @@ export default function App({ children }) {
 
   if (travelClaims.length > 0 && statusesToFilterBy.length === 0) {
     // Sets initial status filters after travelClaims load
-    const initialStatusFilters = [
+    const claimStatusesPresent = [
       ...new Set(travelClaims.map(claim => claim.claimStatus)),
     ];
+    const initialStatusFilters = [];
+
+    // Status order based on sorting of CLAIMS_STATUSES
+    for (const status of Object.keys(CLAIMS_STATUSES)) {
+      if (claimStatusesPresent.includes(status)) {
+        initialStatusFilters.push(status);
+      }
+    }
```

Comment:
Instead of having to map ALL the statuses, which may change and the order is prone to getting messed up, could we rely on just the important statuses and set operations? Like this:

```javascript
import {intersection,difference} from 'lodash';

const topStatuses = new Set(['On Hold', 'Denied', 'In Manual Review']);
const availableStatuses = new Set(travelClaims.map(c => c.claimStatus));

const availableTopStatuses = intersection(
  Array.from(availableStatuses),
  Array.from(topStatuses)
);

const availableNonTopStatuses = difference(
  Array.from(availableStatuses),
  Array.from(topStatuses)
);

const orderedStatusFilters = availableTopStatuses).concat(availableNonTopStatuses);
```

Then we just have to worry about changes to the "important statuses", and we can still float em to the top.

Reply:
It's even nicer if we use the JavaScript-native set ops, but they were only added as of June, and I worry about older browser support.

Reply:
good thought. Updated to this approach with a note to revisit the native options once they're more mature